### PR TITLE
cc-release pier boot state-machine corrections

### PR DESCRIPTION
**Pull request by joemfb**
Some further enhancements to and cleanup of the work done in #1232:

- batch event reads during replay
- remove a superfluous state flag in the pier
- use more specific conditions and assertions for some pier state transitions
- reorder pier initialization, disabling absurdly inefficient explicit log header read

### CODE REVIEW

File: pkg/urbit/vere/pier.c [hunk 1]
```diff
@@ -979,8 +1033,8 @@ _pier_boot_vent(u3_pier* pir_u)
       //
       //    XX this is horrible
       //
-      u3_noun tuf = ( c3__fake == u3h(pir_u->bot) ) ? u3_nul :
-                    u3h(u3t(u3t(u3t(u3t(pir_u->bot)))));
+      u3_noun tuf = ( c3y == pir_u->fak_o ) ? u3_nul :
```
Comment (belisarius222) [inline pkg/urbit/vere/pier.c]: Can you document what this horrible thing is doing, briefly? You seem to understand it at least

File: pkg/urbit/vere/pier.c [hunk 2]
```diff
@@ -1035,11 +1087,15 @@ _pier_boot_ready(u3_pier* pir_u)
...
   c3_assert( u3_psat_init == pir_u->sat_e );
-  c3_assert( c3n == pir_u->liv_o );
-  pir_u->liv_o = c3y;
+
+  if ( ( 0 == god_u) ||
+       ( 0 == log_u) ||
+       (c3y != god_u->liv_o) ||
+       (c3y != log_u->liv_o) )
+  {
+    return;
```
Comment (belisarius222) [inline pkg/urbit/vere/pier.c]: Looks like this now returns, but it used to die. Why the change?

File: pkg/urbit/vere/pier.c (hunk 1 above)
Comment (joemfb) [reply pkg/urbit/vere/pier.c]: Working around vane initialization problems be extracting the ames domains from the boot event, if we're not fake, in order to have the i/o driver ready for any packets that are sent during the boot event. I'm separately working on making ames more resilient to the initialization order so we don't have to do this.

File: pkg/urbit/vere/pier.c (hunk 2 above)
Comment (joemfb) [reply pkg/urbit/vere/pier.c]: I changed the calling convention -- the (two) callers used to verify the preconditions before calling, now the function itself is resilient to being called twice in either order (ie, disk initialized before serf and vice versa) and the call-sites are simpler.

File: pkg/urbit/vere/pier.c (hunk 1 above)
Comment (belisarius222) [reply pkg/urbit/vere/pier.c]: Ok. A comment to this effect would be helpful.

**Comment by joemfb** [comment]
I'll come back and add a comment if I can't remove it. More incoming shortly.
